CMakeLists.txt

```diff
@@ -258,6 +258,8 @@ OCV_OPTION(WITH_CAP_IOS "Enable iOS video capture" ON
   VERIFY HAVE_CAP_IOS)
 OCV_OPTION(WITH_CAROTENE "Use NVidia carotene acceleration library for ARM platform" (NOT CV_DISABLE_OPTIMIZATION)
   VISIBLE_IF (ARM OR AARCH64) AND NOT IOS AND NOT XROS)
+OCV_OPTION(WITH_HALRVV071 "Use rvv 0.7.1 optimizations" (NOT CV_DISABLE_OPTIMIZATION)
```

Comment:
Let's omit all 0.7.1 mentions from everywhere except the source code (e.g. `WITH_HAL_RVV`), we can make universal HAL for RISC-V RVV which would implement algorithms for different RVV versions and toolchains depending on need.

Headers would have tree structure then, for example:

- 3rdparty/hal_rvv/
  - CMakeLists.txt
  - hal_rvv.hpp
  - version/
    - hal_rvv_071.hpp
    - hal_rvv_10.hpp
    - ...